src/calibrationlib/calibration_benchmark_runner.cpp

```diff
@@ -0,0 +1,104 @@
+#include "calibration_benchmark_runner.hpp"
+#include <benchmark_runner.hpp>
+#include <file_based_benchmark_item_runner.hpp>
+#include <file_based_table_generator.hpp>
+#include <tpcds/tpcds_table_generator.hpp>
+#include <tpch/tpch_benchmark_item_runner.hpp>
+#include <tpch/tpch_table_generator.hpp>
+#include "hyrise.hpp"
+
+namespace opossum {
+
+CalibrationBenchmarkRunner::CalibrationBenchmarkRunner(const std::string& path_to_dir)
+    : _feature_export(OperatorFeatureExport(path_to_dir)), _table_export(TableFeatureExport(path_to_dir)) {
+  auto config = std::make_shared<BenchmarkConfig>(BenchmarkConfig::get_default_config());
+  config->max_runs = 1;
+  config->enable_visualization = false;
+  config->chunk_size = 100'000;
+  config->cache_binary_tables = true;
+  _config = config;
+}
+
+CalibrationBenchmarkRunner::CalibrationBenchmarkRunner(const std::string& path_to_dir,
+                                                       std::shared_ptr<BenchmarkConfig> config)
+    : _feature_export(OperatorFeatureExport(path_to_dir)), _table_export(TableFeatureExport(path_to_dir)) {
+  _config = config;
+}
+
+void CalibrationBenchmarkRunner::run_benchmark(const BenchmarkType type, const float SCALE_FACTOR,
+                                               const int number_of_executions) {
+  std::shared_ptr<BenchmarkRunner> benchmark_runner;
+  switch (type) {
+    case BenchmarkType::TCPH:
+      benchmark_runner = _build_tcph(SCALE_FACTOR);
+      break;
+    case BenchmarkType::TCPDS:
+      benchmark_runner = _build_tcpds(SCALE_FACTOR);
+      break;
+    case BenchmarkType::JOB:
+      benchmark_runner = _build_job();
+      break;
+    default:
+      throw std::runtime_error("Provided unknown BenchmarkType.");
+  }
+
+  for (int i = 0; i < number_of_executions; ++i) {
```

Comment:
`i` is not a good name.

Reply:
And we don't use `int` in Hyrise. Probably `size_t` here.